### PR TITLE
feat: adding stages to Snowflake driver

### PR DESCRIPTION
**Pull request by mckraqs**

### CODE REVIEW

**Comment by mckraqs** [comment]
@jravenel @Dr0p42 - PR ready for review

**Comment by jravenel** [comment]
@MinuraPunchihewa @Dr0p42 @FlorentLvr → as mentioned during the roadmap, let's sync together this week to review and document the review process

cc @chicken-biryani

File: naas_drivers/tools/snowflake.py [hunk 1]
```diff
@@ -123,97 +108,116 @@ def connect(
...
-        @param schema: (optional) schema to use for passed query
-        @param role: (optional) role to use for passed query
-        @param n: (optional) query result length limit
+        @param n: query result length limit
         @return: pandas.DataFrame table containing query outcome
         """
-        res = self.query(sql, warehouse, database, schema, role, n)
+        res = self.execute(sql, n)
 
         # TODO: Apply dtypes mapping from ResultMetadata objects
```
Comment (Dr0p42) [inline naas_drivers/tools/snowflake.py:162]: Is this TODO still wanted ?

**Review by Dr0p42 — APPROVED** [review]
Sounds good to me 👍

I like the implementation and the use of multiple classes to keep the code more readable.

The documentation is also very handy and well written which make the overall code easy to understand. I think that someone with a bit of Python knowledge will be able to use this driver without too much difficulty. 

It's, again, a very good contribution @mckraqs 👏

File: naas_drivers/tools/snowflake.py (hunk 1 above)
Comment (mckraqs) [reply naas_drivers/tools/snowflake.py:162]: Yes. Metadata returned from Snowflake about each column contains some information that we can use to create a solid DataFrame out of it, not only pass data itself and hope for proper mapping from Pandas side.

Anyway, I would put it in another issue to fix that. Shouldn't be much of a work, including updating basic notebook example. What do you think @Dr0p42 @jravenel?

Comment (jravenel) [reply naas_drivers/tools/snowflake.py:162]: I think it's a great idea, after v0 we need to think about v1 and talk with businesses. What do you think?

**Pull request merged**